device/src/main/java/se/sigmaconnectivity/blescanner/device/advertiser/BleUidAdvertiserImpl.kt

```diff
@@ -0,0 +1,48 @@
+package se.sigmaconnectivity.blescanner.device.advertiser
+
+import android.bluetooth.BluetoothManager
+import android.bluetooth.le.AdvertiseData
+import android.bluetooth.le.AdvertiseSettings
+import android.bluetooth.le.BluetoothLeAdvertiser
+import android.content.Context
+import android.os.ParcelUuid
+import io.reactivex.Observable
+import se.sigmaconnectivity.blescanner.domain.ble.BleUidAdvertiser
+import se.sigmaconnectivity.blescanner.domain.model.BLEFeatureState
+import se.sigmaconnectivity.blescanner.domain.toHash
+import java.util.*
+
+class BleUidAdvertiserImpl(private val context: Context) : BleAdvertiser(), BleUidAdvertiser {
+
+    override val bluetoothLeAdvertiser: BluetoothLeAdvertiser? by lazy(LazyThreadSafetyMode.NONE) {
+        val bluetoothManager =
+            context.getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
+        bluetoothManager.adapter.bluetoothLeAdvertiser
+    }
+
+    private val settings: AdvertiseSettings = AdvertiseSettings.Builder()
+        .setAdvertiseMode(AdvertiseSettings.ADVERTISE_MODE_LOW_POWER)
+        .setConnectable(false)
+        .setTimeout(0)
+        .setTxPowerLevel(AdvertiseSettings.ADVERTISE_TX_POWER_ULTRA_LOW)
+        .build()
+
+
+    override fun startAdvertising(serviceUUID: String, manufacturerId: Int, userUid: String): Observable<BLEFeatureState> {
```

Comment:
To tylko sugestia. Nie musisz tego wdrażać, ale myślę, że lepiej byłoby nie otwierać startAdvertising. Raczej oczekiwane działanie jest takie samo dla obu potomków (i pewnie pozostałych potencjalnie istnniejących).

może lepiejbyłoby w abstrakcyjnej klasie zaimplementować startAdvertising z generowaniem parametrow (które w klasie abstrakcyjnej byłyby abstrakcyjne)

Natomiast w klasach dziedziczących mielibyśmy implementacje getterów to tych parametrów i to one robiłyby różnicowanie.

Oczywiście to nie robi jakiejść wielkiej duplikacji itd, ale np. ryzykowne jest (z punktu widzenia czystości kodu), że ktoś w potomnej klasie zacznie odpalać te advertismenty zupełnie w inny sposób, nie korzystając z tego co jest w klasie potomnej do zmiany statusu itd. A skoro to już znajduje się w klasie macierzystej, a w.g. mnie nie chcielibyśmy tego.